From d3cd8c491559ca9eb7ce81242df3b3927466e6d9 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 26 May 2021 15:56:18 +0900
Subject: [PATCH] cifsd: simplify error handling in
 ksmbd_gen_preauth_integrity_hash()
Git-commit: d3cd8c491559ca9eb7ce81242df3b3927466e6d9
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Simplify error handling in ksmbd_gen_preauth_integrity_hash().

Reviewed-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/auth.c | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/fs/cifsd/auth.c b/fs/cifsd/auth.c
index b0a9e4591fa5..1d4c4e6d28cd 100644
--- a/fs/cifsd/auth.c
+++ b/fs/cifsd/auth.c
@@ -1063,14 +1063,13 @@ int ksmbd_gen_preauth_integrity_hash(struct ksmbd_conn *conn, char *buf,
 	int msg_size = be32_to_cpu(rcv_hdr->smb2_buf_length);
 	struct ksmbd_crypto_ctx *ctx = NULL;
 
-	if (conn->preauth_info->Preauth_HashId ==
-	    SMB2_PREAUTH_INTEGRITY_SHA512) {
-		ctx = ksmbd_crypto_ctx_find_sha512();
-		if (!ctx) {
-			ksmbd_debug(AUTH, "could not alloc sha512 rc %d\n", rc);
-			goto out;
-		}
-	} else {
+	if (conn->preauth_info->Preauth_HashId !=
+	    SMB2_PREAUTH_INTEGRITY_SHA512)
+		return -EINVAL;
+
+	ctx = ksmbd_crypto_ctx_find_sha512();
+	if (!ctx) {
+		ksmbd_debug(AUTH, "could not alloc sha512 rc %d\n", rc);
 		goto out;
 	}
 
-- 
2.33.0

